core/src/main/scala/io/finch/ToService.scala

```diff
@@ -94,19 +98,33 @@ object ToService {
     def apply(
         endpoints: Endpoint[A] :: ET,
         includeDateHeader: Boolean,
-        includeServerHeader: Boolean): Service[Request, Response] = new Service[Request, Response] {
+        includeServerHeader: Boolean,
+        metrics: Metrics): Service[Request, Response] = new Service[Request, Response] {
 
       private[this] val underlying = endpoints.head.handle(respond400OnErrors)
 
       def apply(req: Request): Future[Response] = underlying(Input.fromRequest(req)) match {
-        case EndpointResult.Matched(rem, out) if rem.route.isEmpty => out.map(oa => conformHttp(
-          oa.toResponse(trA, trE),
-          req.version,
-          includeDateHeader,
-          includeServerHeader
-        )).run
-
-        case _ => tsT(endpoints.tail, includeDateHeader, includeServerHeader)(req)
+        case EndpointResult.Matched(rem, out) if rem.route.isEmpty =>
+          val path = underlying.toString
+          val start = Stopwatch.timeMillis().toFloat
+          metrics.countCall(path)
+          out.map(oa => conformHttp(
+            oa.toResponse(trA, trE),
+            req.version,
+            includeDateHeader,
+            includeServerHeader
+          )).run.onSuccess(response => {
+            val now = Stopwatch.timeMillis().toFloat
+            metrics.countSuccess(path, response.statusCode)
+            metrics.successTime(path, response.statusCode, now - start)
+            metrics.size(path, response.content.length.toFloat)
```

Comment:
How does this size hold up when using streaming? Wouldn't this either force the stream, or be inaccurate?

Reply:
AFAIK for streams it's gonna be `0`